### ISSUE TITLE
Expose Encode & Decode Utility Function to Get Cursor

### ISSUE THREAD

**Issue by willjleong** [pilagod/gorm-cursor-paginator#6]
I've been using this library along with `gqlgen` to implement graphql cursor pagination for the graphql relay `connection` standard. I've currently duplicated the priavte encode and decode utility functions in my own util package to provide gorm-cursor-paginator compatible cursors each edge that is resolved through dataloaden (go dataloader implementation) (I use gorm-cursor paginator to select only ID and CreatedAt from the database, return an array of IDs and dataloader resolves so I need to be able to use the encode function outside of the direct pagination). This change would be making `Encode` and `Decode` public methods.

Here's and example of the graphql query. As you can see the standard allows the client to get the cursor for each item in the paginated struct array. Exposing the `Encode` will allow the paginator object to handle the Encoding of these cursors if I resolve the objects through dataloader.

```graphql
{
  user {
    id
    name
    friends(first: 10, after: "opaqueCursor") {
      edges {
        cursor
        node {
          id
          name
        }
      }
      pageInfo {
        hasNextPage
      }
    }
  }
}
```

**Comment by willjleong** [comment]
Yeah pretty much. What you pointed to above. The relay graphql standard has a field in the list of edges that can resolve a cursor for each record (called a node in graphql). I ended up having to copy and reuse code that is in your `utils` folder that I have to hardcode and make sure my cursor composite is the same as what I used for cursor pagination. With graphql, you typically allow `resolvers` to batch fetch the collection by pk ids so I'm making a query like this with your pagination, and then in an edge resolver, I need to get the cursor for each result. 

```golang
func (r *membersConnectionResolver) Edges(ctx context.Context, obj *models.MembersConnection) ([]*models.MembersEdge, error) {
	var results []*models.MembersEdge
	members, errs := dataloader.CtxLoaders(ctx).UserByID.LoadAll(obj.Ids)
	for _, err := range errs {
		if err != nil {
			return nil, gqlerror.Errorf("Failed to Load Organization Members")
		}
	}

	// Todo: refactor this to be more dry when implementing more connections
	for i := 0; i < len(members); i++ {
		cursor := pagination.EncodeBase64([]string{pagination.Convert(members[i].CreatedAt), pagination.Convert(members[i].ID)})
		results = append(results, &models.MembersEdge{
			Cursor: cursor,
			Node:   members[i],
		})
	}
	return results, nil
}
```

**Comment by pilagod** [comment]
Much clearer, thanks for your detailed explanation !

I think export `encode` and `decode` methods would be a good idea in regard to integrating paginator into different situations. 

For now `encode` is depending on `keys` in paginator set by `SetKeys`. In order to publish it, signature of `encode` should be modified:

```go
func Encode(v reflect.Value, keys []string) string {
	fields := make([]string, len(keys))
	for index, key := range keys {
		// ...
	}
	return encodeBase64(fields)
}
```

The second parameter `keys` indicates which properties to encode on struct `v`. After publishing, usage in your case will look like:

```go
func (r *membersConnectionResolver) Edges(ctx context.Context, obj *models.MembersConnection) ([]*models.MembersEdge, error) {
	// ...
	for i := 0; i < len(members); i++ {
		cursor := paginator.Encode(members[i], []string{"CreatedAt", "ID"}) 
		results = append(results, &models.MembersEdge{
			Cursor: cursor,
			Node:   members[i],
		})
	}
	return results, nil
}
```

Do you think this new signature for `Encode` is ok : ) ? (`Decode` will not change since it is not depending on any properties in paginator)

**Comment by willjleong** [comment]
Yup, this seems reasonable and would make it easier to utilize. Thank you!

**Comment by pilagod** [comment]
@willjleong I have released a new patch v1.1.1 to cover this issue : )
